util/conditions/experimental/merge_strategies.go

```diff
@@ -0,0 +1,408 @@
+func getPriority(condition metav1.Condition, negativePolarityConditionTypes sets.Set[string]) mergePriority {
+	switch condition.Status {
+	case metav1.ConditionTrue:
+		if negativePolarityConditionTypes.Has(condition.Type) {
+			return issueMergePriority
+		}
+		return infoMergePriority
+	case metav1.ConditionFalse:
+		if negativePolarityConditionTypes.Has(condition.Type) {
+			return infoMergePriority
+		}
+		return issueMergePriority
+	case metav1.ConditionUnknown:
+		return unknownMergePriority
+	}
+
+	// Note: this should never happen. In case, those conditions are considered like condition with unknown status.
+	return unknownMergePriority
+}
+
+// aggregateMessages return messages for the aggregate operation.
+func aggregateMessages(conditions []ConditionWithOwnerInfo, n *int, kind string, dropEmpty bool) (messages []string, other int) {
+	// create a map with all the messages and the list of objects reporting the same message.
+	messageObjMap := map[string][]string{}
+	for _, condition := range conditions {
+		if dropEmpty && condition.Message == "" {
+			continue
+		}
+
+		m := fmt.Sprintf("(%s)", condition.Status)
+		if condition.Message != "" {
+			m += fmt.Sprintf(": %s", condition.Message)
+		}
+		if _, ok := messageObjMap[m]; !ok {
+			messageObjMap[m] = []string{}
+		}
+		messageObjMap[m] = append(messageObjMap[m], klog.KRef(condition.OwnerResource.Namespace, condition.OwnerResource.Name).String())
+	}
+
+	// compute the order of messages according to the number of objects reporting the same message.
+	// Note: The message text is used as a secondary criteria to sort messages with the same number of objects.
+	messageIndex := make([]string, 0, len(messageObjMap))
+	for m := range messageObjMap {
+		messageIndex = append(messageIndex, m)
+	}
+
+	sort.SliceStable(messageIndex, func(i, j int) bool {
+		return len(messageObjMap[messageIndex[i]]) > len(messageObjMap[messageIndex[j]]) ||
+			(len(messageObjMap[messageIndex[i]]) == len(messageObjMap[messageIndex[j]]) && messageIndex[i] < messageIndex[j])
+	})
+
+	// Pick the first n messages, decrement n.
+	// For each message, add up to three objects; if more add the number of the remaining objects with the same message.
+	// Count the number of objects reporting messages not included in the above.
+	// Note: we are showing up to three objects because usually control plane has 3 machines, and we want to show all issues
+	// to control plane machines if any,
+	for i := range len(messageIndex) {
+		if *n == 0 {
+			other += len(messageObjMap[messageIndex[i]])
+			continue
+		}
+
+		m := messageIndex[i]
+		allObjects := messageObjMap[messageIndex[i]]
+		if len(allObjects) <= 3 {
+			m += fmt.Sprintf(" from %s", strings.Join(allObjects, ", "))
+		} else {
+			m += fmt.Sprintf(" from %s and %d other %s", strings.Join(allObjects[:3], ", "), len(allObjects)-3, kind)
```

Comment:
```suggestion
			m += fmt.Sprintf(" from %s and %d more %s", strings.Join(allObjects[:3], ", "), len(allObjects)-3, kind)
```
nit (maybe, see similar case above)

Not sure if it's actually important to add the kind. Isn't this usually clear in the context of the condition? E.g. for MachinesReady it's obvious what we are aggregating from

Reply:
fixed